ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/AdminNodeInfo.java

```diff
@@ -67,25 +66,23 @@ public JsonRpcResponse response(final JsonRpcRequest req) {
 
     try {
       final Map<String, Object> response = new HashMap<>();
-      final BytesValue nodeId = peerNetwork.getLocalPeerInfo().getNodeId();
-      final InetSocketAddress address = peerNetwork.getDiscoverySocketAddress();
-      final int port = peerNetwork.getLocalPeerInfo().getPort();
+      final Map<String, Integer> ports = new HashMap<>();
 
-      final InetAddress inetAddress = address.getAddress();
-      response.put(
-          "enode",
-          "enode://"
-              + nodeId.toString().substring(2)
-              + "@"
-              + InetAddresses.toUriString(inetAddress)
-              + ":"
-              + port);
+      final PeerInfo peerInfo = peerNetwork.getLocalPeerInfo();
+      final BytesValue nodeId = peerInfo.getNodeId();
+      final Peer advertisedPeer = peerNetwork.getAdvertisedPeer();
+      if (advertisedPeer != null) {
```

Comment:
Should we update `getAdvertisedPeer` to return an `Optional`?

Reply:
Done.  Good idea.